ibis/expr/typing.py

```diff
@@ -0,0 +1,9 @@
+""" Define types for annotation """
+
+from typing import Optional, Tuple, Union
+
+import pandas as pd
+
+# Time context types
+SupportsTimestamp = Union[pd.Timestamp, str]
+TimeContext = Optional[Tuple[SupportsTimestamp, SupportsTimestamp]]
```

Comment:
its slightly more clear to use Optional[TimeContext] everywhere (as we do with AggContext), i know it longer but more idiomatic :-<

Reply:
Got it, I have made all TimeContext appearances to be Optiona[TimeContext]